### PR TITLE
feature/Issue #865 sort recommended topics

### PR DESCRIPTION
**Pull request by tomy-pg**

### CODE REVIEW

File: src/selectors/topics/sort_topic_list.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import { Topic } from './topic';
```
Comment (rasmus-storjohann-PG) [inline src/selectors/topics/sort_topic_list.ts]: So can this file be deleted now?

File: src/selectors/topics/types.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    readonly id: string;
+    readonly title: string;
+    readonly description: string;
+    readonly taxonomyTerms: ReadonlyArray<TaxonomyTermReference>;
+    readonly exploreSection: ExploreSection;
+    readonly isRecommended: boolean;
+    readonly relatedTopics: ReadonlyArray<TopicListItem>;
+}
+
+export interface TopicListItem {
```
Comment (rasmus-storjohann-PG) [inline src/selectors/topics/types.ts]: Rename to `RelatedTopic`.

File: src/selectors/topics/types.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    readonly id: string;
+    readonly title: string;
+    readonly description: string;
+    readonly isRecommended: boolean;
+    readonly exploreSection: ExploreSection;
+}
+
+export type ExploreTopic = Topic | TopicListItem;
+
+export type ExploreTopicList = ReadonlyArray<Topic> | ReadonlyArray<TopicListItem>;
```
Comment (rasmus-storjohann-PG) [inline src/selectors/topics/types.ts]: Create new interface with the properties that the selector code needs, and change the selector code to use that. Remove these two Explore types

File: src/selectors/topics/select_topic_for_current_explore_section.ts [hunk 4]
```diff
@@ -14,8 +14,9 @@ import { pickTopics } from './pick_topics';
 import { sortTopicList } from './sort_topic_list';
 import { getAllTaxonomyIdsFromAnswers } from '../questionnaire/get_all_taxonomy_ids_from_questionnaire';
 import { pickAnswers } from '../questionnaire/pick_answers';
+import { ExploreTopicList} from './types';
 
-export const selectTopicForCurrentExploreSection = (appStore: Store, routerProps: RouterProps): ReadonlyArray<Topic> => {
+export const selectTopicForCurrentExploreSection = (appStore: Store, routerProps: RouterProps): ExploreTopicList=> {
```
Comment (rasmus-storjohann-PG) [inline src/selectors/topics/select_topic_for_current_explore_section.ts]: Keep the return type the way it was.

File: src/selectors/topics/types.ts (hunk 3 above)
Comment (tomy-pg) [reply src/selectors/topics/types.ts]: Removed the two Explore types. 

However, creating an interface with just the properties the sort function needs did not work as expected. 
I created an interface like the following:

```
interface SelectorTopic {
     readonly id: string;
     readonly isRecommended: boolean;
}

```
and included it as the expected type for the sortTopicList function to return. 

When I tried to use sortTopicList while passing either a `Topic` or `TopicListItem`, I received errors like this: 

```
Type ‘readonly SelectorTopic[]’ is not assignable to type ‘readonly Topic[]’.
 Type ‘SelectorTopic’ is missing the following properties from type ‘Topic’: title, description, taxonomyTerms, exploreSection, relatedTopics
```

To get this error to go away, I would have to change many of the selectors to return `SelectorTopic`. We would be less precise with which type we are expecting in this approach and it does not seem like a great trade off. Instead I chose to create a separate sorting function for `TopicListItems` as well.

File: src/selectors/topics/types.ts (hunk 2 above)
Comment (tomy-pg) [reply src/selectors/topics/types.ts]: After taking another look at this, renaming this to RelatedTopic would make the code more difficult to understand. `TopicListItem` is the the type that we use for a recommended topic, explore topic and a bookmarked topic when rendering a topic as part of our UI.

Comment (rasmus-storjohann-PG) [reply src/selectors/topics/types.ts]: OK

File: src/selectors/topics/types.ts (hunk 3 above)
Comment (rasmus-storjohann-PG) [reply src/selectors/topics/types.ts]: Yeah, Topic would be assignable to SelectorTopic, but not the other way around. I pick up on the fact that you would need to go both ways.

**Pull request merged**